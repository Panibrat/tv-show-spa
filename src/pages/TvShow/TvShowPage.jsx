import React, { Component } from 'react';
import styles from './styles.module.css';
import EpisodesList from '../../components/EpisodesList';

export class TvShowPage extends Component {
    componentDidMount() {
        const { fetchTvShows, fetchEpisodesList } = this.props;
        fetchTvShows();
        fetchEpisodesList();
    }

    render() {
        const { data, isFetching, episodesList } = this.props;
        if (isFetching) {
            return (
                <h1>Fetching...</h1>
            )
        }

        return (
            <div className={ styles.container }>
                <div className={ styles.title}>
                    <h1>{ data.name }</h1>
                </div>
                <div className={ styles.content }>
                    <div className={ styles.coverImage }>
                        <img src={ data && data.image && data.image.original ? data.image.original : '#' }
                             alt={ data.name }
                        />
                    </div>
                    <div className={ styles.info }>
                        <div className={ styles.description }>
                            <h3>Description</h3>
                            <span
                                dangerouslySetInnerHTML={{
                                __html: data.summary
                            }}>
                            </span>
                        </div>
                        <div className={ styles.episodesContainer }>
                            <h3>Episodes</h3>
                            <div className={ styles.episodesList }>
                                <EpisodesList episodesList = { episodesList } />
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
