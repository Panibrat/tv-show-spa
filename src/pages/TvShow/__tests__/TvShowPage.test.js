import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { TvShowPage } from '../TvShowPage';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const props = {
    data: {
        name: 'Name',
        image: {
            original: '#'
        },
        summary: 'Summary',
    },
    episodesList: [],
    isFetching: false,
    tvShowsRequested: jest.fn(),
    episodesListRequested: jest.fn(),
}

describe('TvShowPage Component', () => {
    it('should renders without crashing', () => {
        shallow(<TvShowPage {...props} />);
    });

    it('should renders correctly', () => {
        const wrapper = mount(<TvShowPage {...props} />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
