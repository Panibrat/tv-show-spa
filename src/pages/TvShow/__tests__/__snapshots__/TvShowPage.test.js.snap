// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShowPage Component should renders correctly 1`] = `
<TvShowPage
  data={
    Object {
      "image": Object {
        "original": "#",
      },
      "name": "Name",
      "summary": "Summary",
    }
  }
  episodesList={Array []}
  episodesListRequested={
    [MockFunction] {
      "calls": Array [
        Array [],
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isFetching={false}
  tvShowsRequested={
    [MockFunction] {
      "calls": Array [
        Array [],
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="container"
  >
    <div
      className="title"
    >
      <h1>
        Name
      </h1>
    </div>
    <div
      className="content"
    >
      <div
        className="coverImage"
      >
        <img
          alt="Name"
          src="#"
        />
      </div>
      <div
        className="info"
      >
        <div
          className="description"
        >
          <h3>
            Description
          </h3>
          Summary
        </div>
        <div
          className="episodesContainer"
        >
          <h3>
            Episodes
          </h3>
          <div
            className="episodesList"
          >
            <EpisodesList
              episodesList={Array []}
            >
              <div
                className="container"
              >
                <h1>
                  No episodes yet...
                </h1>
              </div>
            </EpisodesList>
          </div>
        </div>
      </div>
    </div>
  </div>
</TvShowPage>
`;
